src/Access/ExternalUserAccessCheck.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_authentication\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\externalauth\AuthMapInterface;
+
+/**
+ * Checks whether a user is external or not.
+ */
+class ExternalUserAccessCheck implements AccessInterface {
+
+  /**
+   * The external authentication map.
+   *
+   * @var \Drupal\externalauth\Authmap
+   */
+  protected $authMap;
+
+  /**
+   * Constructors the ExternalUserAccessCheck.
+   *
+   * @param \Drupal\externalauth\AuthMapInterface $authMap
+   *   The external authentication map.
+   */
+  public function __construct(AuthMapInterface $authMap) {
+    $this->authMap = $authMap;
+  }
+
+  /**
+   * Checks access.
+   *
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The logged in user key.
+   *
+   * @return \Drupal\Core\Access\AccessResultInterface
+   *   The access result.
+   */
+  public function access(AccountInterface $account): AccessResultInterface {
+    $uid = $account->id();
```

Comment:
`$uid` is only used once, therefore I would not create a variable for it and refactor the code a bit.
There is also no need to use empty(), we can just use:

```php
    if ([] === $this->authMap->getAll($account->id())) {
```
Regarding the missing empty line before the return, as there is no rule in PHPCS, I wouldn't flag this for now but highligh this during the next retro.

As this is not a blocker nor defined in any PHPCS, it's up to the user to decide if he wants to do it or not, it's not mandatory.